src/simulators/extended_stabilizer/gates.hpp

```diff
@@ -169,21 +165,22 @@ U1Sample::U1Sample(double lambda)
     coeff_1 = coeff_1 * root_omega_star;
     if(s_z_quadrant)
     {
-      branches = 
-      {
-        sample_branch_t(coeff_0, Gates::s),
-        sample_branch_t(coeff_1, Gates::z)
-      };
+      gates[0] = Gates::s;
+      gates[1] = Gates::z;
     }
     else
     {
-      branches = 
-      {
-        sample_branch_t(coeff_0, Gates::id),
-        sample_branch_t(coeff_1, Gates::s)
-      };
+      gates[0] = Gates::id;
+      gates[1] = Gates::s;
     }
   }
+  phase_0 = coeff_0/std::abs(coeff_0);
+  phase_1 = coeff_1/std::abs(coeff_1);
```

Comment:
It seems we can have `coeff_?` values of zero and in that cases execution will break. Maybe we want this: 
```suggestion
  phase_0 = std::polar(1.0, std::arg(coeff_0));
  phase_1 = std::polar(1.0, std::arg(coeff_1));
```
With `coeff_? = {0.0, 0.0}` will get `phase_? = {1., 0.}` though. Not sure if that is what we want.

Reply:
Yes, that would be the correct behaviour. I didn't consider this case as if one of the coefficients is zero,we'd be applying a pure Pauli or Clifford which shouldn't require any sampling. But I guess it's always possible for such gates to be entered as `U1` gates in the circuit description?